mypy/messages.py

```diff
@@ -175,7 +175,12 @@ def note_multiline(self, messages: str, context: Context, file: Optional[str] =
     # get some information as arguments, and they build an error message based
     # on them.
 
-    def has_no_attr(self, original_type: Type, typ: Type, member: str, context: Context) -> Type:
+    def has_no_attr(self,
+                    original_type: Type,
```

Comment:
If you pass around a symbol table, one option may be to change the type of this argument to `Union[Type, SymbolTable]` (and rename it).